koordinates/sets.py

```diff
@@ -74,27 +145,204 @@ def _deserialize(self, data, manager):
+            self._manager,
+            target_url,
+            valid_filter_attributes=("data",),
+            valid_sort_attributes=(),
+        )
 
-        # reload myself
-        r = self._client.request("GET", base_url)
+    @is_bound
+    def get_version(self, version_id, expand=[]):
+        """
+        Get a specific version of this set
+        """
+        target_url = self._client.get_url(
+            "SET_VERSION", "GET", "single", {"id": self.id, "version_id": version_id},
+        )
+        return self._manager._get(target_url, expand=expand)
+
+    @is_bound
+    def get_draft_version(self, expand=[]):
+        """
+        Get the current draft version of this set.
+        :raises NotFound: if there is no draft version.
+        """
+        target_url = self._client.get_url(
+            "SET_VERSION", "GET", "draft", {"id": self.id}
+        )
+        return self._manager._get(target_url, expand=expand)
+
+    @is_bound
+    def get_published_version(self, expand=[]):
+        """
+        Get the latest published version of this set.
+        :raises NotFound: if there is no published version.
+        """
+        target_url = self._client.get_url(
+            "SET_VERSION", "GET", "published", {"id": self.id}
+        )
+        return self._manager._get(target_url, expand=expand)
+
+    @is_bound
+    def create_draft_version(self):
+        """
+        Creates a new draft version from this model content.
+
+        If anything in the data object has changed then an import will begin immediately.
+        Otherwise to force a re-import from the previous sources call :py:meth:`koordinates.client.Client.start_import`.
+
+        :rtype: Client
+        :return: the new version
+        :raises Conflict: if there is already a draft version for this set.
+        """
+        target_url = self._client.get_url(
+            "SET_VERSION", "POST", "create", {"id": self.id}
+        )
+        r = self._client.request("POST", target_url, json={})
+        return self._manager.create_from_result(r.json())
+
+    @is_bound
+    def publish(self, version_id=None):
+        """
+        Creates a publish task just for this version, which publishes as soon as any import is complete.
+
+        :return: the publish task
+        :rtype: Publish
+        :raises Conflict: If the version is already published, or already has a publish job.
+        """
+        if not version_id:
+            version_id = self.version.id
+
+        target_url = self._client.get_url(
+            "SET_VERSION", "POST", "publish", {"id": self.id, "version_id": version_id},
+        )
+        r = self._client.request("POST", target_url, json={})
+        return self._client.get_manager(Publish).create_from_result(r.json())
+
+    @is_bound
+    def save(self, with_data=False):
+        """
+        Edits this draft layerversion.
+        # If anything in the data object has changed, cancel any existing import and start a new one.
```

Comment:
as before, I think the 'data object' stuff here is wrong, sets don't have that stuff